test/functional/p2p-segwit.py

```diff
@@ -1396,18 +1396,20 @@ def test_signature_version_1(self):
 
         temp_utxos.pop(0)
 
-        # Update self.utxos for later tests. Just spend everything in
-        # temp_utxos to a corresponding entry in self.utxos
+        # Update self.utxos for later tests by creating two outputs
+        # that consolidate all the coins in temp_utxos.
+        output_value = sum(i.nValue for i in temp_utxos) // 2
+
         tx = CTransaction()
         index = 0
+        tx.vout = [CTxOut(output_value, CScript([OP_TRUE]))] * 2
         for i in temp_utxos:
             # Just spend to our usual anyone-can-spend output
             # Use SIGHASH_SINGLE|SIGHASH_ANYONECANPAY so we can build up
             # the signatures as we go.
             tx.vin.append(CTxIn(COutPoint(i.sha256, i.n), b""))
-            tx.vout.append(CTxOut(i.nValue, CScript([OP_TRUE])))
             tx.wit.vtxinwit.append(CTxInWitness())
-            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_SINGLE|SIGHASH_ANYONECANPAY, i.nValue, key)
+            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_ALL|SIGHASH_ANYONECANPAY, i.nValue, key)
```

Comment:
why this change?

Reply:
It felt wrong to use SIGHASH_SINGLE with an out-of-bounds index.

Reply:
ah, ok then comment could use an update a few lines up.

Reply:
Oops!  Thanks, will fix.